pandapower/toolbox.py

```diff
@@ -1801,13 +1802,18 @@ def adapt_element_idx_references(net, element, element_type, offset=0):
                       for ix in net2["line_geodata"].index]
                 net2.line_geodata.set_index(np.array(ni), inplace=True)
             ignore_index = element not in ("bus", "res_bus", "bus_geodata", "line_geodata", "group")
+            ignore_index = ignore_index & (not retain_original_indices_in_net1)
```

Comment:
not a nice peace of code. My proposal:
add `and not retain_original_indices_in_net1` directly at the end of ìgnore_index` declaration.